### ISSUE TITLE
[BUG] High cpu consumption when downloading a distro

### ISSUE THREAD

**Issue by Felipe-Linhares** [bostrot/wsl2-distro-manager#116]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

notice that the video stopped.
https://user-images.githubusercontent.com/70000056/226991348-b9a0d6d5-5529-4cd1-bb0d-d7d539e058d6.mp4

### Expected Behavior

download the distro without crashing the machine

### Steps To Reproduce

1. open wsl manager.
2. I went to add an instance.
3. I put the name.
4. I selected the distro I wanted.
5. I went to create.

### Environment

```markdown
- Windows Version: Windows 11 Pro 22H2 (22621.14.13)
- WSL Distro Manager Version: 1.6.1
- WSL Version: 1.1.3.0
- Source: Windows Store
```

**Comment by bostrot** [comment]
Could you share your hardware specs?

**Comment by Felipe-Linhares** [comment]
- RAM - 20GB
- IdeaPad 3 15ALC6 - AMD Ryzen 5 5500U
- SSD - NVMe IM2P33F4 256GB
- Free space - 50GB

**Comment by bostrot** [comment]
Hmm, that shouldn't be a problem..
Could you try restarting your pc, maybe ending background processes, and check if it is still that slow for you?

**Comment by Felipe-Linhares** [comment]
I did that, but even so, the high CPU consumption and the crash continued, I bought the version from the microsoft store, but it is outdated, it is at 1.6.0 :c

**Comment by bostrot** [comment]
That is a bug. That version shows 1.6.0 but in fact is 1.6.1. I will fix that very soon :) 

Can you also tell me what your Internet bandwidth is? (trying to find a way to reproduce your issue)

**Comment by Felipe-Linhares** [comment]
Aaaaa, got it, my internet is 600Mb.

**Comment by bostrot** [comment]
Does this also happen whem you select other distros? E.g. Ubuntu 22.04?

**Comment by Felipe-Linhares** [comment]
it happens, I tested it with some Ubuntu available, and also Debian

**Comment by bostrot** [comment]
Can you try either version v1.6.0 (from the releases page) or the build [here](https://github.com/bostrot/wsl2-distro-manager/tree/cb9c0a7d74b43d8884ff2ca993aabc5a07061819) (build/windows/runner/release)?

Sorry, to make you do all this but I can't reproduce it :/

**Comment by Felipe-Linhares** [comment]
Okay, I just tested it in version 1.5.0 and 1.6.0, and the same error happens, it crashes the machine

- default cpu usage
![image](https://user-images.githubusercontent.com/70000056/227027599-637c77f5-7d37-4d01-8584-d9624128fb59.png)

- 1.5.0
![image](https://user-images.githubusercontent.com/70000056/227026508-dc617bc0-3212-41c3-a16f-ad0e00c5988b.png)

- 1.6.0
![image](https://user-images.githubusercontent.com/70000056/227027384-40594a86-2fe1-4755-b4d9-2bcd9747fd7f.png)

**Comment by bostrot** [comment]
So the `WSL2 Manager` process makes the usage go to 100% or is it some other process (invoked by wsl manager maybe)?

**Comment by Felipe-Linhares** [comment]
It doesn't appear because the task manager hangs, everything hangs in fact, then I can't identify the process that is increasing the CPU usage

**Comment by bostrot** [comment]
I see. 

Does it also happen when you manually import a rootfs with wsl (without WSL Manager)?

e.g. with [Ubuntu 20.04 rootfs](https://cloud-images.ubuntu.com/releases/focal/release/ubuntu-20.04-server-cloudimg-amd64-wsl.rootfs.tar.gz): `wsl --import testimport C:\WSL2-Distros ~\Downloads\ubuntu-20.04-server-cloudimg-amd64-wsl.rootfs.tar.gz`

**Comment by Felipe-Linhares** [comment]
everything went well, it didn't crash the system

**Comment by bostrot** [comment]
Not sure what is going wrong there then..

I saw that you checked out the source. Can you try to debug it and check somewhere along these lines where exactly it happens: https://github.com/bostrot/wsl2-distro-manager/blob/main/lib/dialogs/create_dialog.dart#L172

I suspect it might have something to do with the SharedPreferences as the only other thing there is probably the call to the import command and that should work as expected.

**Comment by Felipe-Linhares** [comment]
you can leave it, I'll debug it here, if I can get something to help in any way I can, thank you for your time.

**Comment by bostrot** [comment]
Sure, tell me when you find something. I am curious.

Also just for references: the cpu usage on my machine is under 1% for this app when creating a distro.

**Comment by Felipe-Linhares** [comment]
is your machine AMD?

**Comment by bostrot** [comment]
I tried both on Intel and AMD machines.

**Comment by bostrot** [comment]
Just thought about something else: have you tried to disable any antivirus you have?

**Comment by Felipe-Linhares** [comment]
I tried here but got the same result :C

**Comment by bostrot** [comment]
I had hoped that would help more :/

Could try to use the windows diagnose start and see if that changes anything? (win+r then msconfig the second option)

![image](https://user-images.githubusercontent.com/7342321/227342115-4161dd32-71ba-4d89-8c68-87cb0eb150d0.png)

**Comment by Felipe-Linhares** [comment]
i tried but without success

**Comment by lesmo** [comment]
I'm finding this issue too on a Microsoft Surface Studio laptop with i7 11th gen and 32gb of RAM. The process didn't crash, but it slowed the entire PC down horribly.

I also have a Razer Blade 2019 (i7 8th gen, 64gb of RAM) I can test on, but that'll have to wait a bit cuz' I'm out of town for the week.

**Comment by bostrot** [comment]
@lesmo Do you have, by any chance, windows Ransomware protection on?

Also could you too check the task manager and see what process exactly is ramping up?

I guess you are on version 1.7.1?

**Comment by bostrot** [comment]
I just created a build with extra logging. Can you try it again here? It should report back any errors (hopefully):

https://github.com/bostrot/wsl2-distro-manager/suites/12259396937/artifacts/648996275

**Comment by bostrot** [comment]
Is this fixed with v1.7.4?

**Comment by bostrot** [comment]
Closing for now. Reopen if you still encounter this issue.
